app/modules/users/user_controller.py

```diff
@@ -28,6 +30,12 @@ async def get_conversations_for_user(
             repo_name = projects[0].repo_name
             branch_name = projects[0].branch_name
 
+            agent_id = conversation.agent_ids[0] if conversation.agent_ids else None
+
+            custom_agent = self.sql_db.query(CustomAgent).filter_by(id=agent_id).first()
+            display_agent_id = (
+                custom_agent.role if custom_agent else conversation.agent_ids[0]
+            )
             response.append(
```

Comment:
this wil do 100 db calls for 100 conversations, update this logic to do a single db call outside the loop

Reply:
is it possible to do a join between conversation and agent here and get this done in a single query ?